clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10623,7 +10623,7 @@ def err_second_argument_to_cwsc_not_pointer : Error<
   "second argument to __builtin_call_with_static_chain must be of pointer type">;
 
 def err_vector_incorrect_num_elements : Error<
-  "%select{too many|too few}0 elements in vector %select{initialization|operand}3 (expected %1 elements, have %2)">;
+  "%select{too many|too few|incorrect number of}0 elements in vector %select{initialization|operand}3 (expected %1 elements, have %2)">;
```

Comment:
I don't like this change. this should stay being a diagnostic about the number of elements  but in SemaHLSL you are using it to say not a vector

your code:
```
if (!VTy) {
      SemaRef.Diag(TheCall->getBeginLoc(),
                   diag::err_vector_incorrect_num_elements)
```

Reply:
@inbelic [suggested](https://github.com/llvm/llvm-project/pull/125319#discussion_r1941619629) using `diag::err_vec_builtin_non_vector` instead.
It has the message "%select{first two|all}1 arguments to %0 must be vectors"

Reply:
I would still need this change for the other error message when a uint3 is passed.
We don't know if the programmer intended to use the uint2 or the uint4 version of the function, so uint3 has neither more or less elements than expected -- it's just an incorrect number.

Reply:
this diagnostic is not about vector size its just about argument size  to the builtin

Reply:
My latest commit ([28b7eb8](https://github.com/llvm/llvm-project/pull/125319/commits/28b7eb8c6ac603ef1c316e7731b07cb6f6f3a34e)) changed the diag message. Is it now to your liking, or do you still think the diag message should be changed?

Reply:
I'd like to see this change reverted. you are using the wrong diagnostic.

Reply:
I made a new commit ([d2b315a](https://github.com/llvm/llvm-project/pull/125319/commits/d2b315a24f850c6115675ee99c804873f19a8f40)) that creates a new diag message specifically for AddUint64, as per your suggestion (which I can't seem to find anymore?).
Is this better?